tests/unit/maze_transformer/test_tokenizers.py

```diff
@@ -117,6 +119,80 @@ def test_to_ascii(tok_mode):
     ), "ASCII encoding from token ids failed"
 
 
+@mark.parametrize(
+    "maze_ascii, tok_mode",
+    [
+        param(
+            maze_ascii,
+            tok_mode,
+            id=f"{tok_mode.name}_{(len(maze_ascii[0])-1)//2}x{(len(maze_ascii[0])-1)//2}",
+        )
+        for maze_ascii, tok_mode in product(
+            [
+                [
+                    "#######",
+                    "#S    #",
+                    "#X### #",
+                    "#X# # #",
+                    "#X# ###",
+                    "#XXE  #",
+                    "#######",
+                ],
+                [
+                    "#####################",
+                    "#   #       #       #",
+                    "# # # # ### # # #####",
+                    "# #   #   #   # #   #",
+                    "# ####### ##### # # #",
+                    "# #E      #     # # #",
+                    "###X# ########### # #",
+                    "#XXX# #           # #",
+                    "#X##### ########### #",
+                    "#X#     #         # #",
+                    "#X# ######### ### # #",
+                    "#X#         #   # # #",
+                    "#X######### # # ### #",
+                    "#XXXXS#     # #     #",
+                    "# ########### #######",
+                    "# #         #   #   #",
+                    "# # ####### ### # ###",
+                    "# # #       #   #   #",
+                    "# # # ####### ##### #",
+                    "#   #               #",
+                    "#####################",
+                ],
+            ],
+            [
+                TokenizationMode.AOTP_UT_uniform,
+                TokenizationMode.AOTP_UT_rasterized,
+            ],
+        )
+    ],
+)
+def test_maze_to_tokens_roundtrip(maze_ascii: list[str], tok_mode: TokenizationMode):
```

Comment:
only thing I can think of is that it would be good to test that tokenization of these mazes equals some hand-inspected string